### ISSUE TITLE
UnusedPrivateMethod false positive

### ISSUE THREAD

**Issue by ylexus** [pmd/pmd#1764]
**Affects PMD Version:** 
6.13.0

**Rule:**
UnusedPrivateMethod 

**Description:**
Same symptoms as in #1181

**Code Sample demonstrating the issue:**

```
final class MyClass {
    public Object create() {
        return method(new MyInterfaceImpl());
    }

    private Object method(MyInterface delegate) {
        return "blah" + delegate;
    }
}

public interface MyInterface {
    Object interfaceMethod(Object action);
}

final class MyInterfaceImpl implements MyInterface {
    @Override
    public Object interfaceMethod(Object command) {
        return new Object();
    }
}

```

Results in:
```
<pmd xmlns="http://pmd.sourceforge.net/report/2.0.0"
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
     xsi:schemaLocation="http://pmd.sourceforge.net/report/2.0.0 http://pmd.sourceforge.net/report_2_0_0.xsd"
     version="6.13.0" timestamp="2019-04-08T17:08:31.624">
    <file name="package.MyClass.java">
        <violation beginline="8" endline="8" begincolumn="20" endcolumn="47" rule="UnusedPrivateMethod" ruleset="Best Practices"
                   package="package" class="MyClass" method="method"
                   externalInfoUrl="https://pmd.github.io/pmd-6.13.0/pmd_rules_java_bestpractices.html#unusedprivatemethod" priority="3">
            Avoid unused private methods such as 'method(MyInterface)'.
        </violation>
    </file>
</pmd>
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
maven-pmd-plugin 3.11.0

(copied form https://github.com/pmd/pmd/issues/1181#issuecomment-480898285 as I suspect this may be a different issue)

**Comment by jsotuyod** [comment]
Thanks for the report. 2 questions here:

1. Is the plugin configured to enable `typeResolution` info? https://maven.apache.org/plugins/maven-pmd-plugin/pmd-mojo.html#typeResolution
2. Which maven goal are you running to generate these results? `pmd:pmd` or using `verify`?

**Comment by ylexus** [comment]
> 1. Is the plugin configured to enable `typeResolution` info? https://maven.apache.org/plugins/maven-pmd-plugin/pmd-mojo.html#typeResolution
We don't specify typeResolution, so it must be defaulting to "true".
> 2. Which maven goal are you running to generate these results? `pmd:pmd` or using `verify`?
pmd:check

**Comment by jsotuyod** [comment]
Ok, please try using `verify` and let me know if it still reports this violation.

**Comment by ylexus** [comment]
Actually we bound pmd:check to "validate" Re-bound it to "verify" and ran verify and it works. Is this because it needs compiled binaries?

**Comment by ylexus** [comment]
Actually only this projects binds pmd:check away from its default "validate". This explains the mystery of why this only happens in that particular project! I wonder if the plugin can be made to catch such scenarios and fail early?

**Comment by jsotuyod** [comment]
@ylexus yes, type resolution needs the compiled sources and dependencies.

As for how to improve the Maven plugin… @adangel is the man to ask.

I'm closing this for the time being, as PMD itself is working correctly.
